framework/web/UrlRule.php

```diff
@@ -96,6 +96,14 @@ class UrlRule extends Object implements UrlRuleInterface
      * @since 2.0.10
      */
     public $normalizer;
+    /**
+     * @var bool Value used when pattern contains only optional params. If `true` rule pattern will contain slash at
+     * the end of param pattern (last optional param can not be omitted if all other patterns are not omitted), if `false`
+     * rule pattern will contain slash at the beginning of param pattern (first optional param can not be omitted
+     * if all other patterns are not omitted).
+     * @since 2.0.12
+     */
+    public $trailingSlashForDefaults = false;
```

Comment:
I'm going to remove this property. It brings unnecessary complications in the API, and the same effect can be achieved with only one additional rule.